pandas/core/exchange/from_dataframe.py

```diff
@@ -0,0 +1,527 @@
+import ctypes
+import re
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Tuple,
+    Union,
+)
+
+import numpy as np
+
+import pandas as pd
+from pandas.core.exchange.dataframe_protocol import (
+    Buffer,
+    Column,
+    ColumnNullType,
+    DataFrame as DataFrameXchg,
+    DtypeKind,
+)
+from pandas.core.exchange.utils import (
+    ArrowCTypes,
+    Endianness,
+)
+
+_NP_DTYPES: Dict[DtypeKind, Dict[int, Any]] = {
+    DtypeKind.INT: {8: np.int8, 16: np.int16, 32: np.int32, 64: np.int64},
+    DtypeKind.UINT: {8: np.uint8, 16: np.uint16, 32: np.uint32, 64: np.uint64},
+    DtypeKind.FLOAT: {32: np.float32, 64: np.float64},
+    DtypeKind.BOOL: {8: bool},
+}
+
+
+def from_dataframe(df, allow_copy=True):
+    """
+    Build a ``pd.DataFrame`` from any DataFrame supporting the interchange protocol.
+
+    Parameters
+    ----------
+    df : DataFrameXchg
+        Object supporting the exchange protocol, i.e. `__dataframe__` method.
+    allow_copy : bool, default: True
+        Whether to allow copying the memory to perform the conversion
+        (if false then zero-copy approach is requested).
+
+    Returns
+    -------
+    pd.DataFrame
+    """
+    if isinstance(df, pd.DataFrame):
+        return df
+
+    if not hasattr(df, "__dataframe__"):
+        raise ValueError("`df` does not support __dataframe__")
+
+    return _from_dataframe(df.__dataframe__(allow_copy=allow_copy))
+
+
+def _from_dataframe(df: DataFrameXchg, allow_copy=True):
+    """
+    Build a ``pd.DataFrame`` from the DataFrame exchange object.
+
+    Parameters
+    ----------
+    df : DataFrameXchg
+        Object supporting the exchange protocol, i.e. `__dataframe__` method.
+    allow_copy : bool, default: True
+        Whether to allow copying the memory to perform the conversion
+        (if false then zero-copy approach is requested).
+
+    Returns
+    -------
+    pd.DataFrame
+    """
+    pandas_dfs = []
+    for chunk in df.get_chunks():
+        pandas_df = protocol_df_chunk_to_pandas(chunk)
+        pandas_dfs.append(pandas_df)
+
+    if not allow_copy and len(pandas_dfs) > 1:
+        raise RuntimeError(
+            "To join chunks a copy is required which is forbidden by allow_copy=False"
+        )
+    if len(pandas_dfs) == 1:
+        pandas_df = pandas_dfs[0]
+    else:
+        pandas_df = pd.concat(pandas_dfs, axis=0, ignore_index=True, copy=False)
```

Comment:
```suggestion
        pandas_df = pd.concat(pandas_dfs, axis=0, ignore_index=True, copy=allow_copy)
```
Should this be `allow_copy`?

Reply:
Kind of useless, as `allow_copy == False` already (ensured at lines 81-84), but hampers understanding.

Reply:
Makes sense